### PR TITLE
Cleanup AppVersionInfo and turn it into a singleton

### PR DESCRIPTION
**Pull request by cmcfarlen**
This should fix the linking issues with appVersionInfo having to be defined all over.

### CODE REVIEW

**Comment by cmcfarlen** [comment]
[approve ci centos]

**Comment by cmcfarlen** [comment]
[approve ci autest]

**Comment by bneradt** [comment]
[approve ci autest]

**Comment by cmcfarlen** [comment]
[approve ci autest]

File: src/records/P_RecCore.cc [hunk 1]
```diff
@@ -35,6 +35,7 @@
 #include "swoc/bwf_std.h"
 
 #include <fstream>
+#include <iterator>
```
Comment (ywkaras) [inline src/records/P_RecCore.cc:38]: Why do you need this?

**Review by ywkaras — APPROVED** [review]

**Pull request merged**